bitnami/ghost/templates/deployment.yaml

```diff
@@ -1,197 +1,189 @@
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "ghost.fullname" . }}
-              key: ghost-password
-        - name: GHOST_EMAIL
-          value: {{ .Values.ghostEmail | quote }}
-        - name: BLOG_TITLE
-          value: {{ .Values.ghostBlogTitle | quote }}
-        {{- if .Values.smtpHost }}
-        - name: SMTP_HOST
-          value: {{ .Values.smtpHost | quote }}
-        {{- end }}
-        {{- if .Values.smtpPort }}
-        - name: SMTP_PORT
-          value: {{ .Values.smtpPort | quote }}
-        {{- end }}
-        {{- if .Values.smtpUser }}
-        - name: SMTP_USER
-          value: {{ .Values.smtpUser | quote }}
-        {{- end }}
-        {{- if .Values.smtpPassword }}
-        - name: SMTP_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "ghost.fullname" . }}
-              key: smtp-password
-        {{- end }}
-        {{- if .Values.smtpFromAddress }}
-        - name: SMTP_FROM_ADDRESS
-          value: {{ .Values.smtpFromAddress | quote }}
-        {{- end }}
-        {{- if .Values.smtpService }}
-        - name: SMTP_SERVICE
-          value: {{ .Values.smtpService | quote }}
-        {{- end }}
-        ports:
-        - name: http
-          containerPort: 2368
-        {{- if .Values.livenessProbe.enabled }}
-        livenessProbe:
-          httpGet:
-            path: {{ .Values.ghostPath }}
-            port: http
-            httpHeaders:
-            - name: Host
-              value: {{ include "ghost.host" . | quote }}
-            {{- if eq .Values.ghostProtocol "https" }}
-            - name: X-Forwarded-Proto
-              value: https
+        - name: {{ include "ghost.fullname" . }}
+          image: {{ include "ghost.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          env:
+            - name: ALLOW_EMPTY_PASSWORD
+              value: {{ ternary "yes" "no" .Values.allowEmptyPassword | quote }}
+            - name: MARIADB_HOST
+              value: {{ ternary (include "ghost.mariadb.fullname" .) .Values.externalDatabase.host .Values.mariadb.enabled | quote }}
+            - name: MARIADB_PORT_NUMBER
+              value: {{ ternary "3306" .Values.externalDatabase.port .Values.mariadb.enabled | quote }}
+            - name: GHOST_DATABASE_NAME
+              value: {{ ternary .Values.mariadb.db.name .Values.externalDatabase.database .Values.mariadb.enabled | quote }}
+            - name: GHOST_DATABASE_USER
+              value: {{ ternary .Values.mariadb.db.user .Values.externalDatabase.user .Values.mariadb.enabled | quote }}
+            - name: GHOST_DATABASE_PASSWORD
+            {{- if .Values.mariadb.enabled }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "ghost.mariadb.fullname" . }}
+                  key: mariadb-password
+            {{- else }}
+              value: {{ .Values.externalDatabase.password | quote }}
```

Comment:
Didn't see it was set like that. We wouldn't want to have plain text passwords in the yaml. In other charts what we do is to create a secret with the external db data. With that, you can use a helper that would return the correct secret name. I think we do it in Kong

Reply:
yes, you are completely right!